File: algorithm/1018.py
perfectA = [
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
]

perfectB = [
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
    ["B", "W", "B", "W", "B", "W", "B", "W"],
    ["W", "B", "W", "B", "W", "B", "W", "B"],
]

N, M = input().split(" ")
N = int(N)
M = int(M)
board = []
for i in range(0, int(N)):
    board.append(list(input()))

answer = []

countA = 0
countB = 0
for i in range(0, N-7):
    for j in range(0, M-7):
        for k in range(0, 8):
            for l in range(0, 8):
                if board[i + k][j+l] != perfectA[k][l]:
                    countA = countA + 1
                if board[i + k][j+l] != perfectB[k][l]:
                    countB = countB + 1
        answer.append(countA)
        answer.append(countB)
        countA = 0
        countB = 0

answer.sort()
print(answer[0])

File: algorithm/1018.js
// js 빠른 input

const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");

const [n, m] = input.shift().split(" ").map(Number);
input.pop();
const board = input.map((str) => str.split(""));
const boardWhite = [
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
];
const boardBlack = [
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
  ["B", "W", "B", "W", "B", "W", "B", "W"],
  ["W", "B", "W", "B", "W", "B", "W", "B"],
];

const checkBnW = (idxR, idxC) => {
  let [countW, countB] = [0, 0];

  for (let i = idxR; i < idxR + 8; i++) {
    for (let j = idxC; j < idxC + 8; j++) {
      if (board[i][j] === boardWhite[i - idxR][j - idxC]) {
        countW++;
      } else if (board[i][j] === boardBlack[i - idxR][j - idxC]) {
        countB++;
      }
    }
  }

  return Math.min(countW, countB);
};

let min = 3000;
for (let i = 0; i < n - 7; i++) {
  for (let j = 0; j < m - 7; j++) {
    let tmp = checkBnW(i, j);
    min = Math.min(min, tmp);
  }
}

console.log(min);
